### PR TITLE
[docs] Localization progress, polish

### PR DESCRIPTION
**Pull request by oliviertassinari**
A take on this, first time experiencing the badges.

Preview: https://deploy-preview-9672--material-ui-x.netlify.app/x/react-data-grid/localization/

Before

Screenshot 2023-07-14 at 01 44 41

After

Screenshot 2023-07-14 at 01 44 26

Before

Screenshot 2023-07-14 at 01 44 48

After

Screenshot 2023-07-14 at 01 44 35

### CODE REVIEW

File: docs/src/modules/components/LocalizationTable.js [hunk 1]
```diff
@@ -18,50 +17,43 @@ const Root = styled('div')(({ theme }) => ({
...
         ? (theme.vars || theme).palette.text.primary
         : (theme.vars || theme).palette.error.dark,
-
     '& .progress-bar': {
       backgroundColor: (theme.vars || theme).palette.error.main,
       opacity: 0.3,
     },
     '& .progress-background': {
-      border: `1.5px solid ${(theme.vars || theme).palette.error.dark}`,
+      border: `1px solid ${(theme.vars || theme).palette.error.light}`,
```
Comment (noraleonte) [inline docs/src/modules/components/LocalizationTable.js:30]: I love how the lighter versions of the border look on the dark theme. However, I would keep the dark borders on light theme, it looks more consistent with the text. WDYT?
```suggestion
      border: `1px solid`,
      borderColor:
        theme.palette.mode === 'dark'
          ? (theme.vars || theme).palette.error.light
          : (theme.vars || theme).palette.error.dark,
```

File: docs/src/modules/components/LocalizationTable.js [hunk 2]
```diff
@@ -18,50 +17,43 @@ const Root = styled('div')(({ theme }) => ({
...
         : (theme.vars || theme).palette.warning.dark,
-
     '& .progress-bar': {
       backgroundColor: (theme.vars || theme).palette.warning.main,
-      opacity: 0.3,
+      opacity: theme.palette.mode === 'dark' ? 0.4 : 0.25,
     },
     '& .progress-background': {
-      border: `1.5px solid ${(theme.vars || theme).palette.warning.dark}`,
+      border: `1px solid ${(theme.vars || theme).palette.warning.light}`,
```
Comment (noraleonte) [inline docs/src/modules/components/LocalizationTable.js:43]:
```suggestion
      border: `1px solid`,
      borderColor:
        theme.palette.mode === 'dark'
          ? (theme.vars || theme).palette.warning.light
          : (theme.vars || theme).palette.warning.dark,
```

File: docs/src/modules/components/LocalizationTable.js [hunk 3]
```diff
@@ -18,50 +17,43 @@ const Root = styled('div')(({ theme }) => ({
...
     '& .progress-bar': {
       backgroundColor: (theme.vars || theme).palette.success.main,
       opacity: 0.3,
     },
     '& .progress-background': {
-      border: `1.5px solid ${green[200]}`,
-      '&.completed': {
-        border: `1.5px solid ${(theme.vars || theme).palette.success.dark}`,
-      },
+      border: `1px solid ${(theme.vars || theme).palette.success.light}`,
```
Comment (noraleonte) [inline docs/src/modules/components/LocalizationTable.js:56]:
```suggestion
      border: `1px solid`,
      borderColor:
        theme.palette.mode === 'dark'
          ? (theme.vars || theme).palette.success.light
          : (theme.vars || theme).palette.success.dark,
```

**Review by noraleonte — COMMENTED** [review]
Nice improvement! I left a very small nitpick, but love the direction! 🎉

**Review by noraleonte — APPROVED** [review]
Nice! 🚀

File: docs/src/modules/components/LocalizationTable.js (hunk 1 above)
Comment (oliviertassinari) [reply docs/src/modules/components/LocalizationTable.js]: I have tried to use `.main` but it felt still too much distracting from what I think matters: the % progress.

**Pull request merged**